modules/mod_ginger_rdf/controllers/controller_hydra.erl

```diff
@@ -0,0 +1,154 @@
+    RdfResults = [m_rdf_export:to_rdf(Id, [State#state.ontology], Context) || Id <- Result],
+
+    HydraCollectionUrl = hydra_url(page(RequestArgs, undefined), Context), %% The Hydra collection.
+    HydraViewUrl = hydra_url(page(RequestArgs, Page), Context), %% The paged view on the Hydra collection.
+    #rdf_resource{
+        id = HydraCollectionUrl,
+        triples = [
+            #triple{
+                subject = HydraCollectionUrl,
+                predicate = rdf_property:rdf(<<"type">>),
+                object = rdf_property:hydra(<<"Collection">>)
+            },
+            #triple{
+                subject = HydraCollectionUrl,
+                predicate = rdf_property:hydra(<<"totalItems">>),
+                object = #rdf_value{value = Total}
+            },
+            #triple{
+                subject = HydraCollectionUrl,
+                predicate = rdf_property:hydra(<<"view">>),
+                object = HydraViewUrl
+            },
+            #triple{
+                subject = HydraViewUrl,
+                predicate = rdf_property:hydra(<<"first">>),
+                object = hydra_url(page(RequestArgs, 1), Context)
+            },
+            #triple{
+                subject = HydraViewUrl,
+                predicate = rdf_property:hydra(<<"next">>),
+                object = hydra_url(page(RequestArgs, Next), Context)
+            },
+            #triple{
+                subject = HydraViewUrl,
+                predicate = rdf_property:hydra(<<"previous">>),
+                object = hydra_url(page(RequestArgs, Previous), Context)
+            },
+            #triple{
+                subject = HydraViewUrl,
+                predicate = rdf_property:hydra(<<"last">>),
+                object = hydra_url(page(RequestArgs, Pages), Context)
+            }
+            |
+            [#triple{
+                subject = HydraCollectionUrl,
+                predicate = rdf_property:hydra(<<"member">>),
+                object = RdfResult
+            } || RdfResult <- RdfResults]
+        ]
+    }.
+
+%% @doc Serialize RDF resource into one of the RDF serialization formats.
+-spec serialize(#rdf_resource{}, json_ld | turtle) -> binary().
+serialize(RdfResource, json_ld) ->
+    JsonLd = ginger_json_ld:serialize_to_map(RdfResource),
+    jsx:encode(JsonLd);
+serialize(RdfResource, turtle) ->
+    ginger_turtle:serialize(RdfResource).
+
+hydra_url(undefined, _) ->
+    undefined;
+hydra_url(QueryParams, Context) ->
+    z_dispatcher:url_for(hydra, [{use_absolute_url, true} | QueryParams], undefined, Context).
+
+page(_, false) ->
+    undefined;
+page(_, 0) ->
+    undefined;
+page(QueryParams, undefined) ->
+    proplists:delete("page", QueryParams);
+page(QueryParams, PageNumber) when is_number(PageNumber) andalso PageNumber >= 1 ->
+    [{"page", PageNumber} | proplists:delete("page", QueryParams)].
+
+id(Context) ->
+    m_rsc:rid(z_context:get_q(id, Context), Context).
```

Comment:
Maybe if this is named `get_rid_from_q_id` or something (`resource_id_from_q`?), it would have been clearer for me what was happening in the calling code above. Not too important, because here it is very clear what it does.

Reply:
Yeah, that was my (lazy) consideration: because the function body is so small, it should be quite clear what it does. Still, you make a valid point to have function names a tad more expressive than merely `id`.